fedimint-testing/src/bin/fixtures.rs

```diff
@@ -60,14 +62,76 @@ fn bitcoin_rpc() -> anyhow::Result<Arc<BitcoinClient>> {
     Ok(client)
 }
 
-async fn run_bitcoind(state: MutableState) -> anyhow::Result<()> {
+async fn fedimint_client() -> anyhow::Result<UserClient> {
+    let workdir: PathBuf = env::var("FM_CFG_DIR").unwrap().parse()?;
+    let cfg_path = workdir.join("client.json");
+    let db_path = workdir.join("client.db");
+    let cfg: UserClientConfig = load_from_file(&cfg_path).expect("Failed to parse config");
+    let db = fedimint_rocksdb::RocksDb::open(db_path)?;
+    let decoders = module_decode_stubs();
+    let db = Database::new(db, module_decode_stubs());
+    let module_gens = ClientModuleGenRegistry::from(vec![
+        DynClientModuleGen::from(WalletClientGen),
+        DynClientModuleGen::from(MintClientGen),
+        DynClientModuleGen::from(LightningClientGen),
+    ]);
+    Ok(Client::new(cfg.clone(), decoders, module_gens, db, Default::default()).await)
+}
+
+async fn record_pid(process: &Child) -> anyhow::Result<()> {
+    let pid_file = env::var("FM_PID_FILE").unwrap();
+    let mut file = OpenOptions::new()
+        .write(true)
+        .append(true)
+        .open(pid_file)
+        .await?;
+
+    let mut buf = Vec::<u8>::new();
+    writeln!(buf, "{}", process.id().expect("PID missing"))?;
+    file.write_all(&buf).await?;
+
+    Ok(())
+}
+
+async fn await_bitcoin_rpc(waiter_name: &str) -> anyhow::Result<()> {
+    let rpc = bitcoin_rpc()?;
+
+    while rpc.get_blockchain_info().is_err() {
+        sleep(Duration::from_secs(1)).await;
+        info!("{waiter_name} waiting for bitcoin rpc ...");
+    }
+
+    Ok(())
+}
+
+async fn await_fedimint_block_sync() -> anyhow::Result<()> {
+    // FIXME: make sure we've had at least 101 blocks ... probably best to call
+    // await_bitcoin_rpc and do it there? but then the function has weird name ...
```

Comment:
Could pass a closure or something to await_bitcoin_rpc that would encapsulate that logic. Then not all callers of `await_bitcoin_rpc` would need to wait 101 blocks if its not necessary. Won't block on this though

Reply:
Since `run_bitcoind` mines 101 blocks as soon as it starts, I think we should just wait for 101 blocks to be mined in all cases. That won't take long, and it's a nice invariant. For example, then we can send from the default rpc wallet (`pegin.sh` needs this) and LND won't blow up when it tries to start (if bitcoin RPC commands return that they are still loading the block index, it does blow up).